### PR TITLE
first run at building the binary into a container

### PR DESCRIPTION
**Pull request by jdambly**
created entry script
create makefile
create Dockerfile

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
@jdambly Thanks for this PR! Is this mostly focused on building distributing the bond-cni binary inside a cluster ala https://github.com/k8snetworkplumbingwg/sriov-cni/blob/master/images/k8s-v1.16/sriov-cni-daemonset.yaml
?

File: Makefile [hunk 1]
```diff
@@ -1,10 +1,49 @@
...
+GOCLEAN=$(GOCMD) clean
+GOTEST=$(GOCMD) test
+GOGET=$(GOCMD) get
+BINARY_NAME=./bin/bond
+
+# the following vars are overidable with ENV vars
+# example:
+#  before running make set vars like so
+#  export DOCKER_REPO=my_repo
+IMAGE_REPO?=jdambly
```
Comment (killianmuldoon) [inline Makefile:14]: These aren't good defaults - it's better to leave this with very generic default values for the repo etc. unless we have a stable known host for the image.
I think we want it to be clear that the image is not available from a public registry

File: Makefile [hunk 2]
```diff
@@ -1,10 +1,49 @@
-deps-update:
+# Go parameters
+ROOT_DIR:=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+GOCMD=go
```
Comment (killianmuldoon) [inline Makefile:3]: I think this Makefile may be overly complex for bond cni - is this based on template?

File: Makefile [hunk 3]
```diff
@@ -1,10 +1,49 @@
...
+	$(GOTEST) -v ./...
+clean: ## remove binaries
+	$(GOCLEAN)
+	rm -f $(BINARY_NAME)
+	rm -f $(BINARY_UNIX)
+deps: ##  go mod <deps>
 	go mod tidy && \
 	go mod vendor
 
+docker-build:
```
Comment (killianmuldoon) [inline Makefile:44]: make image could be cleaner here?

File: images/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  fi
+done
+
+# Copy file into proper place.
+if cp -f "$BON_BIN_FILE" "$CNI_BIN_DIR"; then
+    echo "Bond CNI installed Success!"
+    exit 0
+else
+    echo "Clound not copy file"
+    exit 1
```
Comment (killianmuldoon) [inline images/entrypoint.sh:63]: Missing newline at end of file

File: images/entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    exit 1;
+  fi
+done
+
+# Copy file into proper place.
+if cp -f "$BON_BIN_FILE" "$CNI_BIN_DIR"; then
+    echo "Bond CNI installed Success!"
+    exit 0
+else
+    echo "Clound not copy file"
```
Comment (killianmuldoon) [inline images/entrypoint.sh:62]: * Could

File: images/entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+for i in $CNI_BIN_DIR $BOND_BIN_FILE
+do
+  if [ ! -e "$i" ]; then
+    /bin/echo "Location $i does not exist"
+    exit 1;
+  fi
+done
+
+# Copy file into proper place.
+if cp -f "$BON_BIN_FILE" "$CNI_BIN_DIR"; then
```
Comment (killianmuldoon) [inline images/entrypoint.sh:58]: BOND_BIN_FILE

File: images/entrypoint.sh [hunk 7]
```diff
@@ -0,0 +1,63 @@
+#!/bin/sh
+
+# Always exit on errors.
+set -e
+
+# Set known directories.
+CNI_BIN_DIR="/host/opt/cni/bin"
```
Comment (killianmuldoon) [inline images/entrypoint.sh:7]: What does this directory refer to here?

File: images/entrypoint.sh [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+BOND_BIN_FILE="/usr/bin/bond"
+
+# Give help text for parameters.
+usage()
+{
+    printf "This is an entrypoint script for SR-BOND CNI to overlay its\n"
```
Comment (killianmuldoon) [inline images/entrypoint.sh:13]: SR- ?

**Comment by bcg62** [comment]
this would be useful. thanks.

**Comment by killianmuldoon** [comment]
@bcg62 What problem are you looking for this PR to solve? I'd be happy to add something like this, but I'd like to understand the use case first.

**Comment by bcg62** [comment]
@killianmuldoon using bond-cni in kubernetes with multus/sr-iov.

multus-cni and sriov-cni have containers and a helm chart https://github.com/k8snetworkplumbingwg/helm-charts these are used to install the plugin binaries. I'd like to install this the same way.

**Comment by killianmuldoon** [comment]
@jdambly are you planning on updating this PR?

File: Makefile (hunk 1 above)
Comment (jdambly) [reply Makefile:14]: sure I can remove this

File: Makefile (hunk 2 above)
Comment (jdambly) [reply Makefile:3]: yes this I was looking at the makefile for the sr-iov plugin. I can trim this stuff out to make it more simple

File: images/entrypoint.sh (hunk 7 above)
Comment (jdambly) [reply images/entrypoint.sh:7]: this is the path for the volume mounts inside the container

**Comment by jdambly** [comment]
@killianmuldoon hey sry about this I prematurely submitted this PR, and got distracted by other things, but I'm back to looking at this again, thanks for the great feedback! I will be making all the fixes you suggested and will update the PR.

The intent of this PR was to streamline installing the CNI into k8s clusters with version > 1.16.x, and match the deployment of this CNI with multus/sr-iov. Once this work is done I also intend to create a helm chart, since my company requires all add-ons to be installed with helm 3

**Comment by Eoghan1232** [comment]
Closing this PR as it's stale.

**Pull request closed without merging**